### PR TITLE
US6429: Basic Welcome Message

### PR DESCRIPTION
**Pull request by lukemoderwell**
This PR does the following:
  - gives CRDS the ability to parse cookie strings into usable key value pairs
  - creates a method that can add a user's name to an `HTML` element
  - adds unit tests for the service

### CODE REVIEW

File: apps/crossroads_interface/test/js/userService_test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  afterEach(() => {
+    document.body.innerHTML = '';
+    user = null;
+    welcomeTitle = null;
+    deleteCookie('username');
+  });
+
+  it('should customize the welcome message based on user name', () => {
+    document.cookie = 'username=Rodney';
+    user = new CRDS.UserService();
```
Comment (mjvcallibrity) [inline apps/crossroads_interface/test/js/userService_test.js]: Are you meaning to create a second new UserService. You already do this once up in the beforeEach().

File: apps/crossroads_interface/web/static/js/userService.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+CRDS.UserService = class UserService {
+  constructor() {
+    this.userId = Cookies.get('userId');
+    this.profileName = Cookies.get('username');
+  }
+
+  customizeName(htmlId, fallbackName) {
+    const el = document.getElementById(htmlId);
+    if (el !== null) {
+      const elText = this.profileName !== undefined ? this.profileName : fallbackName;
```
Comment (mjvcallibrity) [inline apps/crossroads_interface/web/static/js/userService.js]: I realize that it would be really edge case, but it is possible that things could get messed up in Ministry Platform and a user's first name or their nickname could be an empty string. Because you are only checking for undefined, this.profilename could equal an empty string and return that empty string, resulting in a message like "Welcome, !" in the personalized welcome section. We should also probably check against empty strings to avoid this unlikely, but undesirable edge case.

File: apps/crossroads_interface/test/js/userService_test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  it('should fallback if no name is detected', () => {
+    user.customizeName('foo', 'Bodacious Dude');
+    expect(welcomeTitle.innerText).toEqual('Bodacious Dude');
+  });
+
+  it('should not do anything if the ID is not on the page', () => {
+    user.customizeName('beef', 'Fellow Carnivore');
+    expect(welcomeTitle.innerText).toEqual('');
+  });
+});
```
Comment (mjvcallibrity) [inline apps/crossroads_interface/test/js/userService_test.js]: We should add another test case for the case in which the username is an empty string (see comments below in the implementation code).

File: apps/crossroads_interface/test/js/userService_test.js (hunk 1 above)
Comment (lukemoderwell) [reply apps/crossroads_interface/test/js/userService_test.js]: @mjvcallibrity  I add that because the UserService doesn't watch for cookie changes so it will only work if you initialize after cookies have been changed. Should I take it out of the beforeEach() and just add to each `it` block?
